lib/request.coffee

```diff
@@ -322,4 +306,50 @@ module.exports = getRequest = ({
 	# (or resolve to) a new response, or throw/reject.
 	###
 
+	###*
+	# @summary Refresh token on user request
+	# @function
+	# @public
+	#
+	# @description
+	# This function automatically refresh token on user request.
+	# The token is refreshed even if it is not expired yet.
+	# Error is thrown if user is not authorized.
+	#
+	# @param {String} options.url - relative url
+	#
+	# @returns {String} token - new token
+	#
+	# @example
+	# request.refreshToken
+	# 	baseUrl: 'https://api.resin.io'
+	###
+
+	exports.refreshToken = (options = {}) ->
+
+		{ baseUrl } = options
+
+		# Only refresh if we have resin-auth
+		if not (auth?)
+			throw new Error ('Unauthorized')
```

Comment:
NIT: Can we make this a bit more explanatory? Something like `auth module not provided in initializer` or something similar? I would like to differentiate between this error and a 401 unauthorized error